clang/lib/ASTMatchers/ASTMatchFinder.cpp

```diff
@@ -1336,6 +1336,44 @@ class MatchASTVisitor : public RecursiveASTVisitor<MatchASTVisitor>,
     return false;
   }
 
+  template <typename T> static SourceLocation getNodeLocation(const T &Node) {
+    return Node.getBeginLoc();
+  }
+
+  static SourceLocation getNodeLocation(const QualType &Node) { return {}; }
+
+  static SourceLocation getNodeLocation(const NestedNameSpecifier &Node) {
+    return {};
+  }
+
+  static SourceLocation getNodeLocation(const CXXCtorInitializer &Node) {
+    return Node.getSourceLocation();
+  }
+
+  static SourceLocation getNodeLocation(const TemplateArgumentLoc &Node) {
+    return Node.getLocation();
+  }
+
+  static SourceLocation getNodeLocation(const Attr &Node) {
+    return Node.getLocation();
+  }
+
+  bool isInSystemHeader(const SourceLocation &Loc) {
```

Comment:
I'm surprised you take a SourceLocation by ref. Isn't we usually take it by value?